transport/split/writer.go

```diff
@@ -36,10 +38,19 @@ var _ io.ReaderFrom = (*splitWriterReaderFrom)(nil)
 // A write will end right after byte index prefixBytes - 1, before a write starting at byte index prefixBytes.
 // For example, if you have a write of [0123456789] and prefixBytes = 3, you will get writes [012] and [3456789].
 // If the input writer is a [io.ReaderFrom], the output writer will be too.
-func NewWriter(writer io.Writer, prefixBytes int64) io.Writer {
-	sw := &splitWriter{writer, prefixBytes}
-	if rf, ok := writer.(io.ReaderFrom); ok {
-		return &splitWriterReaderFrom{sw, rf}
+// If repeatsNumber > 1, then packets will be split multiple times, skipping skipBytes in between splits.
+// Example:
```

Comment:
So the slice points will be in positions `prefixBytes + i * skipBytes`, for `0 <= i < repeatsNumber`.

It may be worth it writing down that formula. It clarifies things.

This formula made me realize that `skipBytes` and `repeatsNumber` are not meaningful by themselves, and must always be specified together.

So perhaps we need a setter like `SplitWriter.SetMultiSplit(splitBytes, splitCount)`.
It could be other names, like `Enable` instead of `Set`, or `Additional`, `Extra` instead of `Multi`, ... Open to ideas.

I think splitCount in this case should be for the extra blocks, in order to not be confusing. In that case the formula would be:
`prefixBytes + i * splitBytes`, for `0 <= i <= splitCount`. With the default being splitCount == 0.